pkg/pod/pod.go

```diff
@@ -354,6 +359,26 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 	return newPod, nil
 }
 
+// applyTaskLevelComputeResources applies the task-level compute resource requirements to the Pod.
+func applyTaskLevelComputeResources(steps []v1beta1.Step, computeResources *corev1.ResourceRequirements, taskSpec v1beta1.TaskSpec) error {
+	if taskSpec.StepTemplate != nil {
+		taskSpec.StepTemplate.Resources = corev1.ResourceRequirements{}
+	}
+
+	// TODO: apply LimitRanges
+
+	for i := range steps {
+		if i == 0 {
```

Comment:
in the TEP i have the resources applied only to the first container, but I think it would actually be better to split them among the containers. The reason is that limitrange default requests are split among the containers. consider this example:

2 steps
task level memory: 1Gi
limitrange default requests: 600Mi

If you split the task level resources between containers you'll have each container with 500Mi (which is what we want). If you just apply them to one container, the limitrange defaults will kick in for the container without resources and you'll have one container with 1Gi and one with 300Mi (not what we want). 

I'll update the TEP to clarify this.